File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    d = {name:score for name,score in zip(name,yearning)}
    answer=[]
    print(d)
    for i in photo:
        sum =0
        for j in i:
            if j in d:
                sum+=d[j]
        answer.append(sum)     
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    const data = {}
    for(let i in name) {
        data[name[i]] = yearning[i]
    }
     
    return photo.map(
        (arr)=> arr.reduce((acc, cur)=> acc + (data[cur] ||0), 0)
    )
    
    return answer;
}